linkerd/detect/src/lib.rs

```diff
@@ -147,19 +149,21 @@ where
             let mut buf = BytesMut::with_capacity(capacity);
             let detected = match time::timeout(timeout, detect.detect(&mut io, &mut buf)).await {
                 Ok(Ok(protocol)) => {
-                    debug!(?protocol, elapsed = ?time::Instant::now().saturating_duration_since(t0), "DetectResult");
+                    debug!(
+                        ?protocol,
+                        elapsed = ?time::Instant::now().saturating_duration_since(t0),
+                        "Detected protocol",
+                    );
                     Ok(protocol)
                 }
                 Err(_) => Err(DetectTimeoutError(timeout, std::marker::PhantomData)),
                 Ok(Err(e)) => return Err(e),
             };
 
             trace!("Dispatching connection");
-            let svc = inner.new_service((detected, target));
-            let mut svc = svc.ready_oneshot().await.map_err(Into::into)?;
-            svc.call(io::PrefixedIo::new(buf.freeze(), io))
-                .await
-                .map_err(Into::into)?;
+            let svc = inner.new_service(detected);
+            let mut svc = svc.ready_oneshot().await?;
```

Comment:
looks like removing the `map_err(Into::into)` broke compilation here: https://github.com/linkerd/linkerd2-proxy/actions/runs/4166843841/jobs/7211692762